plugins/modules/rhsm_repository.py

```diff
@@ -181,7 +181,7 @@ def repository_modify(module, state, name, purge=False):
             if fnmatch(repo['id'], repoid):
                 matched_existing_repo[repoid].append(repo)
                 # Update current_repo_list to return it as result variable
-                updated_repo_list[idx]['enabled'] = True if state == 'enabled' else False
+                updated_repo_list[idx]['enabled'] = True if state in ['enabled', 'present'] else False
```

Comment:
Hmm, usually enabled is different from present. Here it seems to be about enabling the repo, not about it being present in the list. I guess if `state == 'present'` it shouldn't care whether it's enabled or not.

Reply:
The situation in this case is different: there is nothing you can "add" or "remove", as the list of repositories available as part of the subscription(s) of the system is sent by the subscription server. (Strictly speaking, you can "add or remove repositories" in case your account is in entitlement mode, and you attach or remove subscriptions from it; it is a different operation though, and it is not what this module handles.)

So the only operations you can do with the repositories available via subscriptions are enabling or disabling them; if you notice, also the "negative" options `disabled` and `absent` also work in the same way.

Reply:
Ok, in that case this looks good.